query/node.go

```diff
@@ -101,6 +144,58 @@ type functionExpression struct {
 	groupByCollapses bool
 }
 
+// QueryString does the heavy lifting so implementations don't have to.
+func (f functionExpression) QueryString() string {
+	switch f.functionName {
+	case "+", "-", "*", "/":
+		if len(f.arguments) != 2 {
+			// Then it's not actually an operator.
+			break
+		}
+		return fmt.Sprintf("(%s %s %s)", f.arguments[0].QueryString(), f.functionName, f.arguments[1].QueryString())
+	}
+	argumentQueries := make([]string, len(f.arguments))
+	for i := range argumentQueries {
+		argumentQueries[i] = f.arguments[i].QueryString()
+	}
+	argumentString := strings.Join(argumentQueries, ", ")
+	groupString := ""
+	if len(f.groupBy) != 0 {
+		groupKeyword := "group by"
+		if f.groupByCollapses {
+			groupKeyword = "collapse by"
+		}
+		groupString = fmt.Sprintf(" %s %s", groupKeyword, strings.Join(f.groupBy, ", "))
+	}
+	return fmt.Sprintf("%s(%s%s)", f.functionName, argumentString, groupString)
+}
+func (f functionExpression) Name() string {
+	// TODO: deprecate this behavior before it becomes permanent
+	if f.functionName == "transform.alias" && len(f.arguments) == 2 {
+		log.Printf("I'm an alias")
+		if alias, ok := f.arguments[1].(*stringExpression); ok {
```

Comment:
Where would this case fail? Also let's call `alias.QueryString()` and then we can have the contract that all `QueryString` responses should be properly escaped.